addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -180,124 +198,92 @@ public String getReference() {
     @Override
     public void scan(HttpMessage msg, String param, String value) {
 
-        // Number of targets to try
-        int targetCount = 0;
-
-        // Debug only
         LOGGER.debug("Attacking at Attack Strength: {}", this.getAttackStrength());
-
         // Figure out how aggressively we should test
-        switch (this.getAttackStrength()) {
-            case LOW:
-                // Check only for baseline targets (2 reqs / param)
-                targetCount = 3;
-                break;
-
-            case MEDIUM:
-                // This works out as a total of 9 reqs / param
-                targetCount = 9;
-                break;
-
-            case HIGH:
-                // This works out as a total of 15 reqs / param
-                targetCount = REDIRECT_TARGETS.length;
-                break;
-
-            case INSANE:
-                // This works out as a total of 15 reqs / param
-                targetCount = REDIRECT_TARGETS.length;
-                break;
-
-            default:
-                break;
-        }
+        int payloadCount = getPayloadCount();
 
         LOGGER.debug(
                 "Checking [{}][{}], parameter [{}] for Open Redirect Vulnerabilities",
                 getBaseMsg().getRequestHeader().getMethod(),
                 getBaseMsg().getRequestHeader().getURI(),
                 param);
 
-        // For each target in turn
-        // note that depending on the AttackLevel,
-        // the number of elements that we will try changes.
-        String payload;
         String redirectUrl;
+        int payloadIdx = 0;
 
-        for (int h = 0; h < targetCount; h++) {
-            if (isStop()) {
+        // For each payload in turn
+        // note that depending on the Strength,
+        // the number of elements that we will try changes.
+        for (RedirectPayloads payload : RedirectPayloads.values()) {
+            if (isStop() || ++payloadIdx == payloadCount) {
```

Comment:
I think I made a mistake on the second condition, will verify/adjust in the morning.